@here/harp-map-controls/lib/MapControls.ts

```diff
@@ -1312,25 +1341,22 @@ export class MapControls extends THREE.EventDispatcher {
         return { fromWorld, toWorld };
     }
 
-    private panFromTo(fromX: number, fromY: number, toX: number, toY: number): void {
-        const vectors = this.getWorldPositionWithElevation(fromX, fromY, toX, toY);
-        if (vectors === undefined) {
-            return;
-        }
-        const { fromWorld, toWorld } = vectors;
-
+    private panFromTo(fromWorld: THREE.Vector3, toWorld: THREE.Vector3): void {
         // Cancel zoom inertia if a panning is triggered, so that the mouse location is kept.
         this.stopZoom();
 
+        // Assign the new animation start time.
+        this.m_panAnimationStartTime = performance.now();
+
         if (this.mapView.projection.type === geoUtils.ProjectionType.Planar) {
-            // Assign the new animation start time.
-            this.m_panAnimationStartTime = performance.now();
             this.m_panDistanceFrameDelta.subVectors(fromWorld, toWorld);
-            this.pan();
         } else if (this.mapView.projection.type === geoUtils.ProjectionType.Spherical) {
-            // TODO: HARP-6277: Implement inertia in globe.
-            MapViewUtils.rotateCameraAroundGlobe(this.mapView, fromWorld, toWorld);
+            this.m_lastRotateGlobeFromVector.copy(fromWorld);
+            this.m_lastRotateGlobeAxis.crossVectors(fromWorld, toWorld).normalize();
+            this.m_lastRotateGlobeAngle = Math.acos(fromWorld.normalize().dot(toWorld.normalize()));
```

Comment:
If you are limited to 90 degrees (to angles gives same sinus value between 0-PI), angle may be retrieved directly from cross product of normalized vectors, then divide cross by sines and normalization of cross is done.
Maybe not important, but gives small optimization.